calyx-opt/src/passes/compile_static.rs

```diff
@@ -75,99 +87,265 @@ fn make_guard_dyn(
+        }
+
+        // rewrite static_group[go] to early_reset_group[go]
+        // don't have to worrry about writing static_group[done] b/c static
+        // groups don't have done holes.
+        comp.for_each_assignment(|assign| {
+            assign.for_each_port(|port| {
+                self.group_rewrite
+                    .get(&port.borrow().canonical())
+                    .map(Rc::clone)
+            })
+        });
+
+        comp.get_static_groups_mut().append(sgroups.into_iter());
+
+        Ok(Action::Continue)
+    }
+
+    /// Executed after visiting the children of a [ir::Static] node.
+    fn start_static_control(
+        &mut self,
+        sc: &mut ir::StaticControl,
+        comp: &mut ir::Component,
+        sigs: &ir::LibrarySignatures,
+        _comps: &[ir::Component],
+    ) -> VisResult {
+        // assume that there are only static enables left.
+        // if there are any other type of static control, then error out.
+        match sc {
+            ir::StaticControl::Enable(s) => {
+                let sgroup = s.group.borrow_mut();
+                let sgroup_name = sgroup.name();
+
+                // get the "early reset group". It should exist, since we made an 
+                // early_reset group for every static group in the component 
+                let early_reset_name =
+                    self.reset_early_map.get(&sgroup_name).unwrap_or_else(|| {
+                        panic!("group {} not in self.reset_early_map", sgroup_name)
+                    });
+                // check if we've already built the wrapper group for early_reset_group 
+                // if so, we can just use that, otherwise, we must build the wrapper group 
+                let group_choice = match self.wrapper_map.get(early_reset_name){
+                        None => {
+                            // create the builder/cells that we need to create wrapper group 
+                            let mut builder = ir::Builder::new(comp, sigs);
+                            let fsm_name = self.fsm_map.get(early_reset_name).unwrap_or_else(|| unreachable!("group {} has no correspondoing fsm in self.fsm_map", early_reset_name));
+                            let wrapper = Self::build_wrapper_group(fsm_name, early_reset_name, & mut builder);
+                            self.wrapper_map.insert(*early_reset_name, wrapper.borrow().name());
+                            wrapper
+                        }
+                        Some(name) => {
+                            comp.find_group(*name).unwrap()
+                        }
+                    };
+
+                let mut e = ir::Control::enable(group_choice);
+                let attrs = std::mem::take(&mut s.attributes);
+                *e.get_mut_attributes() = attrs;
+                Ok(Action::Change(Box::new(e)))
+            }
+            _ => unreachable!("Non-Enable Static Control should have been compiled away. Run static-inliner to do this"),
```

Comment:
Don't hard code the name of the pass. Instead, import it and use the `::name()` method on it to generate the correct pass name.